modules/ingester/flush.go

```diff
@@ -165,49 +168,64 @@ func (i *Ingester) flushLoop(j int) {
 			return
 		}
 		op := o.(*flushOp)
+		op.attempts++
+
+		retry := false
 
-		var completeBlockID uuid.UUID
-		var err error
 		if op.kind == opKindComplete {
 			level.Debug(log.Logger).Log("msg", "completing block", "userid", op.userID)
 			instance, exists := i.getInstanceByID(op.userID)
 			if !exists {
 				// instance no longer exists? that's bad, log and continue
-				level.Error(log.Logger).Log("msg", "instance not found", "tenantID", op.userID)
+				level.Error(log.Logger).Log("msg", "instance not found", "userID", op.userID, "block", op.blockID.String())
```

Comment:
is this possible?

should we call `getOrCreateInstance`? if we did that it would just fail on the next step since the block wouldn't exist?

Reply:
Don't think it's possible, but went ahead and more properly handled the error.